### PR TITLE
GraphQL - October 2021

### PR DESCRIPTION
**Pull request by sunli829**
This PR is for #682

- [x] Specified By - [GraphQL - October 2021] #677
  - [x] `Scalar`  procedural macro
  - [x] `NewType`  procedural macro
  - [x] `scalar` macro
  - [x] documentation
- [x] Allow directive on variable definition - [GraphQL - October 2021] #678
  - [x] Parser
  - [x] AST
-  [x] ~Allow interface to implement other interface - [GraphQL - October 2021] #680~
    **has been implemented in `v2.1.12`** #322
- [x] Subscription typename - [GraphQL - October 2021] #681
  - [x] Add validation rule
- [x] Number value literal lookahead restrictions - [GraphQL - October 2021] #685
- [X] changelog

### CODE REVIEW

File: src/types/upload.rs [hunk 1]
```diff
@@ -111,6 +111,7 @@ impl Type for Upload {
             description: None,
             is_valid: |value| matches!(value, Value::String(_)),
             visible: None,
+            specified_by_url: None,
```
Comment (Miaxos) [inline src/types/upload.rs:114]: Maybe we could add the multipart spec? `https://github.com/jaydenseric/graphql-multipart-request-spec`

**Review by Miaxos — COMMENTED** [review]
awesome!

File: src/types/upload.rs (hunk 1 above)
Comment (sunli829) [reply src/types/upload.rs:114]: > Maybe we could add the multipart spec? `https://github.com/jaydenseric/graphql-multipart-request-spec`

done🙂

**Review by Miaxos — APPROVED** [review]
awesome!! 👏

**Comment by numToStr** [comment]
@sunli829 Awesome work! I was wondering, Will this be a major or minor release?

**Comment by sunli829** [comment]
> @sunli829 Awesome work! I was wondering, Will this be a major or minor release?

This is a minor version because `2021` is forward compatible.

**Comment by sunli829** [comment]
Released in `v2.10.0`

**Pull request merged**